avocado/core/runner.py

```diff
@@ -395,8 +394,7 @@ def sigtstp_handler(signum, frame):     # pylint: disable=W0613
             self.job.log.debug('')
 
         self.result.check_test(test_state)
-        if not status.mapping[test_state['status']]:
-            failures.append(test_state['name'])
+        summary[test_state['status']] = test_state['fail_class']
```

Comment:
This could mask some "fail_classes" in case they share the same status. What do you think about the idea of processing the status here and returning the result instead (PASS, FAIL, INTERRUPTED). It'd make sense to me, because the way it works is:

`job -> runner -> test(s) -> runner -> job`

The job does not need to know the details, it's the runner which should either pass or fail.